terraform/environments/mlra/modules/ecs/main.tf

```diff
@@ -494,40 +520,40 @@ resource "aws_cloudwatch_log_stream" "cloudwatch_stream" {
 
 # Added to support target tracking scaling_adjustment
 
-resource "aws_iam_role" "ecs-autoscale-role" {
-  name = "ecs-scale-application"
-
-  assume_role_policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": "application-autoscaling.amazonaws.com"
-      },
-      "Effect": "Allow"
-    }
-  ]
-}
-EOF
-}
+# resource "aws_iam_role" "ecs-autoscale-role" {
+#   name = "ecs-scale-application"
+#
+#   assume_role_policy = <<EOF
+# {
+#   "Version": "2012-10-17",
+#   "Statement": [
+#     {
+#       "Action": "sts:AssumeRole",
+#       "Principal": {
+#         "Service": "application-autoscaling.amazonaws.com"
+#       },
+#       "Effect": "Allow"
+#     }
+#   ]
+# }
+# EOF
+# }
 
 
 # ECS cluster scaling
 
-resource "aws_iam_role_policy_attachment" "ecs-autoscale" {
-  role       = aws_iam_role.ecs-autoscale-role.id
-  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceAutoscaleRole"
-}
+# resource "aws_iam_role_policy_attachment" "ecs-autoscale" {
+#   role       = aws_iam_role.ecs-autoscale-role.id
+#   policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceAutoscaleRole"
+# }
 
 resource "aws_appautoscaling_target" "ecs_target" {
   max_capacity       = var.appscaling_max_capacity
   min_capacity       = var.appscaling_min_capacity
   resource_id        = "service/${aws_ecs_cluster.ecs_cluster.name}/${aws_ecs_service.ecs_service.name}"
   scalable_dimension = "ecs:service:DesiredCount"
   service_namespace  = "ecs"
-  role_arn           = aws_iam_role.ecs-autoscale-role.arn
+  # role_arn           = aws_iam_role.ecs-autoscale-role.arn
 }
```

Comment:
Removing the optional role_arn will set to default Service-Link role, which replaces the [deprecating Service Role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/security-iam-awsmanpol-deprecated-policies.html) AmazonEC2ContainerServiceAutoscaleRole defined previously, which was already [ignored by Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appautoscaling_target#role_arn).